--- etcher-1.0.0~beta.15.orig/scripts/build/linux.sh
+++ etcher-1.0.0~beta.15/scripts/build/linux.sh
@@ -32,11 +32,6 @@ if ! command -v bower 2>/dev/null; then
   exit 1
 fi
 
-if ! command -v upx 2>/dev/null; then
-  echo "Dependency missing: upx" 1>&2
-  exit 1
-fi
-
 if ! command -v python 2>/dev/null; then
   echo "Dependency missing: python" 1>&2
   exit 1
@@ -102,11 +97,6 @@ function package_x86 {
 
   mv $output_package/Etcher $output_package/etcher
   chmod a+x $output_package/*.so*
-
-  # UPX fails for some reason with some other so libraries
-  # other than libnode.so in the x86 build
-  upx -9 $output_package/etcher $output_package/libnode.so
-
 }
 
 function package_x64 {
@@ -126,7 +116,6 @@ function package_x64 {
 
   mv $output_package/Etcher $output_package/etcher
   chmod a+x $output_package/*.so*
-  upx -9 $output_package/etcher $output_package/*.so*
 }
 
 function app_dir_create {
